src/bp/core/services/action/local-action-server.ts

```diff
@@ -86,13 +94,14 @@ export class LocalActionServer {
     this.app.post(
       '/action/run',
       _validateRunRequest,
-      this.asyncMiddleware(async (req, res) => {
-        const { incomingEvent, actionArgs, actionName, botId, token } = req.body
+      this.asyncMiddleware(async (req: RunActionRequest, res: RunActionResponse) => {
+        const { actionArgs, actionName, botId, token, incomingEvent } = req.body
 
         const service = await this.actionService.forBot(botId)
         await service.runLocalAction({ actionName, actionArgs, incomingEvent, token, runType: 'http' })
 
-        res.send({ incomingEvent })
+        const { temp, user, session } = incomingEvent.state
+        res.send({ event: { state: { temp, user, session } } })
```

Comment:
``res.send({event: _.pick(incomingEvent.state, ['temp', 'user', 'session']) })``

Reply:
`_.pick()` is problematic because it does not check if keys exist. 
For example, this compiles just fine:
```ts
 _.pick<IO.EventState>(incomingEvent.state, ['bleh', 'user', 'session'])
```